lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -106,15 +106,27 @@ export class PublicClientApplication {
      * containing data from the server (returned with a null or non-blocking error).
      */
     async handleRedirectCallback(authCallback: AuthCallback): Promise<void> {
-        if(!authCallback) {
+        if (!authCallback) {
             throw BrowserConfigurationAuthError.createInvalidCallbackObjectError(authCallback);
         }
 
         this.authCallback = authCallback;
         const { location: { hash } } = window;
-        if (UrlString.hashContainsKnownProperties(hash)) {
-            const interactionHandler = new RedirectHandler(this.authModule, this.browserStorage, this.authCallback);
-            interactionHandler.handleCodeResponse(hash);
+        const cachedHash = this.browserStorage.getItem(TemporaryCacheKeys.URL_HASH);
+        try {
+            const interactionHandler = new RedirectHandler(this.authModule, this.browserStorage, this.authCallback, this.config.auth.navigateToLoginRequestUrl);
+            let tokenResponse: TokenResponse;
+            if (UrlString.hashContainsKnownProperties(hash)) {
+                tokenResponse = await interactionHandler.handleCodeResponse(hash);
+            } else if (UrlString.hashContainsKnownProperties(cachedHash)) {
+                tokenResponse = await interactionHandler.handleCodeResponse(cachedHash);
+            }
+
+            if (tokenResponse) {
+                this.authCallback(null, tokenResponse);
+            }
+        } catch (err) {
```

Comment:
nitpick: Can we add comments for readability?

Reply:
I will add comments in a separate PR